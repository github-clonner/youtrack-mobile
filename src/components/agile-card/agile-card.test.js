import React from 'react';
import { shallow } from 'enzyme';
import AgileCard from './agile-card';
import toJson from 'enzyme-to-json';

describe('<AgileCard/>', () => {
  let fakeIssue;

  beforeEach(() => {
    fakeIssue = {
      id: 'testIssue',
      idReadable: 'TT-123',
      summary: 'issue summary',
      project: {
        shortName: 'TT'
      },
      fields: [{
        $type: 'jetbrains.charisma.customfields.complex.enumeration.SingleEnumIssueCustomField',
        value: {
          name: 'Critical',
          color: {id: 4, background: '#000', foreground: '#FFF'}
        },
        projectCustomField: {
          field: {
            name: 'Priority',
          }
        }
      }]
    };
  });

  it('should init', () => {
    const wrapper = shallow(<AgileCard issue={fakeIssue} />);
    wrapper.should.be.defined;
  });

  it('should render snapshot', () => {
    const tree = shallow(<AgileCard issue={fakeIssue} />);
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should show summary', () => {
    const wrapper = shallow(<AgileCard issue={fakeIssue} />);
    wrapper.find({testID: 'card-summary'}).children().should.have.text(fakeIssue.summary);
  });
});
