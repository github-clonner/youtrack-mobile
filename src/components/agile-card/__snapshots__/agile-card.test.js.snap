// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AgileCard/> should render snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFF",
        "flexDirection": "column",
        "height": 131,
        "padding": 8,
      },
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <ColorField
        color={
          Object {
            "background": "#000",
            "foreground": "#FFF",
            "id": 4,
          }
        }
        fullText={true}
        style={
          Object {
            "paddingLeft": 4,
            "paddingRight": 4,
            "width": null,
          }
        }
        text="TT-123"
      />
    </View>
  </View>
  <Text
    numberOfLines={3}
    style={
      Object {
        "color": "#444444",
        "fontSize": 13,
        "paddingBottom": 4,
        "paddingTop": 4,
      }
    }
    testID="card-summary"
  >
    issue summary
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "height": 40,
      }
    }
  />
</View>
`;
